src/sage/rings/polynomial/laurent_polynomial.pyx

```diff
@@ -1316,6 +1316,68 @@ cdef class LaurentPolynomial_univariate(LaurentPolynomial):
             return ~self
         raise ArithmeticError("element is not a unit")
 
+    def xgcd(self, other):
+        """
+        Extended `gcd` for univariate Laurent polynomial rings over a field.
+
+        EXAMPLES::
+
+            sage: S.<t> = LaurentPolynomialRing(QQ)
+            sage: (t^-2 + 1).xgcd(t^-3 + 1)
+            (1, 1/2*t^2 - 1/2*t^3 - 1/2*t^4, 1/2*t^3 + 1/2*t^4)
+        """
+        R = self.parent()
+        if not R.is_exact() or not R.base_ring().is_field:
```

Comment:
I had issues in some cases when the base ring is not a field, I can check later for another solution. I tried a test; the error is raised for `RR` but not for `ZZ`. I do not understand.

Reply:
However, those things should be allowed to fail where they actually fail. Trying to catch certain cases ahead of time is just going to be problematic and not futureproof,.

Reply:
What do you propose? Keep only no exact as possible error and let crash eventually for the rest?

Reply:
Just to let it fail, exactly the same for the other case. Your algorithm does not require these conditions; only the one for the underlying polynomials.

Reply:
So, do I eliminate the check and apply no test?  With no check:

- Base ring `ZZ`, apparently works
- Base ring `RR, it works, for sure the result cannot be exact
- Base ring a polynomial ring raise a `NotIMplementedError`
- Bases ring `Zmod(4)`, complete crash:
```
Flint exception (Impossible inverse):
    Cannot invert modulo 2*0
------
...
------
Unhandled SIGABRT: An abort() occurred.
This probably occurred because a *compiled* module has a bug
in it and is not properly wrapped with sig_on(), sig_off().
Python will now terminate.

```